Changes to GettingStartedWithGitHub.md (fixes #176)

johlju opened the pull request:
Added how to resolve merge conflicts
Added how to create a .gitignore file
This resolves issue #176

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/dscresources/177)
<!-- Reviewable:end -->

msftclas commented:
Hi __@johlju__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

mbreakey3 commented:
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 157 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJRZC_zwam3TFilGyU:-KPJRZC_zwam3TFilGyV:bnfttfp) ([raw file](https://github.com/powershell/dscresources/blob/b8642954f2e2c11755da518ca1012b058b7ebbb4/GettingStartedWithGitHub.md#L157)):*
> ```Markdown
> 
> If another pull request is merged while yours are in review, then you have to get those new changes into your pull request before yours are allowed to merge.
> ```

I would add in a link to the [GitHub help page](https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/) on this topic as well

---

*[GettingStartedWithGitHub.md, line 182 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJQX5ZPkdyLs2BlRJY:-KPJQX5ZPkdyLs2BlRJZ:b-jm7ubk) ([raw file](https://github.com/powershell/dscresources/blob/b8642954f2e2c11755da518ca1012b058b7ebbb4/GettingStartedWithGitHub.md#L182)):*
> ```Markdown
> ```
> 
> To fix this we need to manually open the file in a editor of your choosing and find the conflict. You find the conflict by searching for seven equals sign: `=======`.  
> ```

'in an editor'

---

*[GettingStartedWithGitHub.md, line 202 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJQfbJupXbW4m7Ea6C:-KPJQfbJupXbW4m7Ea6D:bx7qhmk) ([raw file](https://github.com/powershell/dscresources/blob/b8642954f2e2c11755da518ca1012b058b7ebbb4/GettingStartedWithGitHub.md#L202)):*
> ```Markdown
> ```
> 
> * Above the equals sign `========` is whats in the `README.md` of your branch.
> ```

'is what's in the'   (and in the line below as well) 

---

*[GettingStartedWithGitHub.md, line 222 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJQr5rqJzvc2OAEHHv:-KPJQr5rqJzvc2OAEHHw:by6w1k) ([raw file](https://github.com/powershell/dscresources/blob/b8642954f2e2c11755da518ca1012b058b7ebbb4/GettingStartedWithGitHub.md#L222)):*
> ```Markdown
> _Note: You must remove the lines `<<<<<<< HEAD`, `========` and `>>>>>>> origin/dev`.
> 
> When your happy with the file. Save it and continue with the next file, if there was more merge conflicts. 
> ```

'When you are happy'

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 157 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJRZC_zwam3TFilGyU:-KPJUy998narJYkA64YD:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b8642954f2e2c11755da518ca1012b058b7ebbb4/GettingStartedWithGitHub.md#L157)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> I would add in a link to the [GitHub help page](https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/) on this topic as well
</details>
Done

---

*[GettingStartedWithGitHub.md, line 182 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJQX5ZPkdyLs2BlRJY:-KPJV-yBB29xBz-JMeZP:b-z321tq) ([raw file](https://github.com/powershell/dscresources/blob/b8642954f2e2c11755da518ca1012b058b7ebbb4/GettingStartedWithGitHub.md#L182)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'in an editor'
</details>
Done. Also changed 'we' to 'you'.

---

*[GettingStartedWithGitHub.md, line 202 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJQfbJupXbW4m7Ea6C:-KPJV6fjIHQvKIyZapQE:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b8642954f2e2c11755da518ca1012b058b7ebbb4/GettingStartedWithGitHub.md#L202)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'is what's in the'   (and in the line below as well) 
</details>
Done

---

*[GettingStartedWithGitHub.md, line 222 \[r2\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJQr5rqJzvc2OAEHHv:-KPJV8_erHTF_nIOg28k:btyizmf) ([raw file](https://github.com/powershell/dscresources/blob/b8642954f2e2c11755da518ca1012b058b7ebbb4/GettingStartedWithGitHub.md#L222)):*
<details><summary><i>Previously, mbreakey3 (Mariah) wrote…</i></summary>
> 'When you are happy'
</details>
Done. Also missed a closing '_' for the italic in the row above.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177)*
<!-- Sent from Reviewable.io -->

mbreakey3 commented:
@kwirkykat - do you have any comments on this? Otherwise, I'm going to merge

KarolKaczmarek commented on GettingStartedWithGitHub.md:
```diff
@@ -149,8 +149,85 @@ git commit -a -m "Update my awesome feature with codereview feedback"
+* Above the equals sign `========` is what's in the `README.md` of your branch.
+* Below the equals sign `========` is what's in the `README.md` of the dev (origin) branch.
+
+To resolve this we have to manually change this section manually. In this example we can do it so the resulting `README.md` looks like this. 
```
You have 'manually' twice

KarolKaczmarek commented on GettingStartedWithGitHub.md:
````diff
@@ -149,8 +149,85 @@ git commit -a -m "Update my awesome feature with codereview feedback"
+```
+_Note: You must remove the lines `<<<<<<< HEAD`, `========` and `>>>>>>> origin/dev`._
+
+When you are happy with the file. Save it and continue with the next file, if there was more merge conflicts. 
````
There should be comma instead of first dot.

KarolKaczmarek commented on GettingStartedWithGitHub.md:
```diff
@@ -149,8 +149,85 @@ git commit -a -m "Update my awesome feature with codereview feedback"
+When you are happy with the file. Save it and continue with the next file, if there was more merge conflicts. 
+
+### To update your pull request
+This is exactly like you usally update the pull request, but using a standard message so it's clear that the commit is for resolving merge conflicts.
```
typo: usally -> usually

kwirkykat commented:
This actually is not the correct procedure for merging new code into a PR anymore.
We are switching over to using ```git rebase``` instead of ```git merge```.

@johlju I can post the correct git rebasing process in this file tomorrow.
Thanks for bringing my attention to this :+1:

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 160 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPKpVewpOOnlGczELus:-KPKpVexHhhmwe8grGiF:b-74bdif) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L160)):*
> ```Markdown
> 
> It's a three step process. 
> 1. Get changes from base branch
> ```

In the DSC Resource Kit, contributors should use forks rather than their own branches.
There should also be instructions here for updating your fork's dev branch with the changes from the remote dev branch.
(https://help.github.com/articles/syncing-a-fork/)

Also, contributors should use **rebasing** instead of merging.

This is what the process actually should look like:
1. Configure tracking of the original repository's dev branch if you have not done so already
2. Fetch changes from the upstream dev branch
3. Rebase your fork's dev branch onto the upstream dev branch
4. Fix any conflicts between your fork's dev and the upstream dev
5. Force push your fork's dev branch
(6.) If you have a separate working branch in your fork, rebase your working branch onto your fork's dev branch
(7.) If you have a separate working branch in your fork, fix any conflicts between your working branch and your fork's dev branch
(8.) If you have a separate working branch in your fork, force push to your working branch

The pull request will then be updated automatically.
It does not require an extra commit for your merge, and the branch history will now be correct.
The changes you made for your PR will be 'replayed' on top of the changes that were recently merged into dev as though you originally created your branch/fork from the current point that the dev branch is at.

---

*[GettingStartedWithGitHub.md, line 171 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPKondi_aXhgJKf81J7:-KPKondj2ZIw2rv10wF_:b-39g0zd) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L171)):*
> ```Markdown
> git checkout <your PR branch>       # Checkout (move) to your pull request branch, i.e git checkout awesome_feature
> git fetch origin dev                # Get all changes from origin/dev
> git merge origin/dev                # Merge changes from origin/dev into your pull request branch.
> ```

This should actually be
```git rebase origin dev```
Fix your merge conflicts...
Then
```
git add <files you fixed merge conflicts in>
git rebase --continue
git push origin <working branch> --force
```

---

*[GettingStartedWithGitHub.md, line 228 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPKxVi9R91t_MTxiGyi:-KPKxVi9R91t_MTxiGyj:b8w4d2m) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L228)):*
> ```Markdown
> This is exactly like you usally update the pull request, but using a standard message so it's clear that the commit is for resolving merge conflicts.
> ```
> git commit -a -m "Resolved merge conflicts" 
> ```

See comment above. Extra commit not necessary with rebasing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177#-:-KPKy6K8WmmaDJ_SKewc:bi3q2nt)*
<!-- Sent from Reviewable.io -->

johlju commented:
I will fix all the comments this afternoon/evening

johlju commented:
Okey, so this was a big change... Please proof read all again. :smile: 

---

Review status: 0 of 1 files reviewed at latest revision, 6 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 160 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPKpVewpOOnlGczELus:-KPOaAcPWNHuZXtDLIrN:b-duy3ua) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L160)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> In the DSC Resource Kit, contributors should use forks rather than their own branches.
> There should also be instructions here for updating your fork's dev branch with the changes from the remote dev branch.
> (https://help.github.com/articles/syncing-a-fork/)
> 
> Also, contributors should use **rebasing** instead of merging.
> 
> This is what the process actually should look like:
> 1. Configure tracking of the original repository's dev branch if you have not done so already
> 2. Fetch changes from the upstream dev branch
> 3. Rebase your fork's dev branch onto the upstream dev branch
> 4. Fix any conflicts between your fork's dev and the upstream dev
> 5. Force push your fork's dev branch
> (6.) If you have a separate working branch in your fork, rebase your working branch onto your fork's dev branch
> (7.) If you have a separate working branch in your fork, fix any conflicts between your working branch and your fork's dev branch
> (8.) If you have a separate working branch in your fork, force push to your working branch
> 
> The pull request will then be updated automatically.
> It does not require an extra commit for your merge, and the branch history will now be correct.
> The changes you made for your PR will be 'replayed' on top of the changes that were recently merged into dev as though you originally created your branch/fork from the current point that the dev branch is at.
</details>
Done. I hope. This was above my git-fu. :sweat_smile:  But I did try it out on my PR, so it seems to work. But please be thorough when reviewing! :smile:  

---

*[GettingStartedWithGitHub.md, line 171 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPKondi_aXhgJKf81J7:-KPObFXlWoDyqtFmQZLr:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L171)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> This should actually be
> ```git rebase origin dev```
> Fix your merge conflicts...
> Then
> ```
> git add <files you fixed merge conflicts in>
> git rebase --continue
> git push origin <working branch> --force
> ```
> 
> 
</details>
Done

---

*[GettingStartedWithGitHub.md, line 206 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJCL2j_oij3-RaV_oh-r3-206:-KPObHvWkZTOmHKovdxd:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L206)):*
<details><summary><i>Previously, KarolKaczmarek (Karol Kaczmarek) wrote…</i></summary>
> You have 'manually' twice
</details>
Done

---

*[GettingStartedWithGitHub.md, line 223 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJCL2j_oij3-RaV_oh-r3-223:-KPObJAyMoDGA0iIVkGP:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L223)):*
<details><summary><i>Previously, KarolKaczmarek (Karol Kaczmarek) wrote…</i></summary>
> There should be comma instead of first dot.
</details>
Done

---

*[GettingStartedWithGitHub.md, line 226 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPJCL2j_oij3-RaV_oh-r3-226:-KPObNiSSqZDtjN6skzR:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L226)):*
<details><summary><i>Previously, KarolKaczmarek (Karol Kaczmarek) wrote…</i></summary>
> typo: usally -> usually
</details>
Done

---

*[GettingStartedWithGitHub.md, line 228 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPKxVi9R91t_MTxiGyi:-KPObWW5Yi_lOzByY3YI:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L228)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> See comment above. Extra commit not necessary with rebasing.
</details>
Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177#-:-KPObY5M01imdsWCsCGJ:bimlc4w)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
@johlju Nice update! :+1: Just a couple wording changes and taking out the extra 'tracking' step, and then this is good to go

---

Review status: 0 of 1 files reviewed at latest revision, 24 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 171 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPKondi_aXhgJKf81J7:-KPouTwo08zyMxIQbEiV:bazpiyi) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L171)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done
</details>
Wow. I just realized that origin in this documentation is actually the upstream branch...
Normally git defaults to setting up origin as your fork, so I always think of 'origin' as your fork.
Sorry if that caused any confusion.


The ```git branch -u origin/dev``` step is not needed.

---

*[GettingStartedWithGitHub.md, line 110 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp1QsPVQloefc5r3o5:-KPp1QsPVQloefc5r3o6:bq0842a) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L110)):*
> to your forks dev branch
to your fork's dev branch

---

*[GettingStartedWithGitHub.md, line 111 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPowWduDW1V2n52oK3o:-KPowWdvUM1FuB-VEC4W:b-wtwa5k) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L111)):*
> ```Markdown
> * To make changes, create a new local branch: `git checkout -b <branch> my/dev`, i.e. `git checkout -b awesome_feature my/dev`. 
> This will create a new local branch, connect to your forks dev branch as the tracking branch (upstream branch) and then checkout (move) to the new branch.  
> _Note: The tracking branch will later be used for resolving merge conflicts_ 
> ```

This note is not needed. You don't need to set up a tracking branch to resolve merge conflicts.

---

*[GettingStartedWithGitHub.md, line 158 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp-UtK64q7ufsfpeGT:-KPp-UtK64q7ufsfpeGU:b-we3yr7) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L158)):*
>  merged while yours are in review,
merged while yours is in review,
> then you have to get those new changes into
you will need to add those new changes into
> before your pull request are allowed to merge
before your pull request is allowed to merge

---

*[GettingStartedWithGitHub.md, line 159 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp031auZMi9cHRe07D:-KPp031auZMi9cHRe07E:b-pqiw6) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L159)):*
> That
This

---

*[GettingStartedWithGitHub.md, line 162 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0ITcF9rlfCZ0Jrnb:-KPp0ITcF9rlfCZ0Jrnc:b-551olm) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L162)):*
> It's a few steps to get this done.
Here are the steps to rebasing your branch:

---

*[GettingStartedWithGitHub.md, line 181 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0XUdxN-Kj5r18Qq5:-KPp0XUdxN-Kj5r18Qq6:b-gx2n1y) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L181)):*
> needs
need

---

*[GettingStartedWithGitHub.md, line 183 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0bXFIRoDYoLcT6TO:-KPp0bXFIRoDYoLcT6TP:b-v0xa71) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L183)):*
> your fork dev
your fork's dev

---

*[GettingStartedWithGitHub.md, line 194 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoyHb9AgcQX_9wKEaO:-KPovsRvZj_-lsTGzT9i:b-nq3ret) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L194)):*
> ```Markdown
> cd <path to cloned repository>      # This is the path to your cloned repository. I.e. cd C:\Source\xActiveDirectory.
> git checkout <your PR branch>       # Checkout (move) to your working branch, i.e git checkout awesome_feature.
> git branch -u my/dev                # If not already tracking the your forks dev branch when you created the branch, this will make sure. _This sets the tracking (upstream) branch._
> ```

Again ```git branch -u origin/dev``` step is not needed.

---

*[GettingStartedWithGitHub.md, line 197 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0iHPx0z0ImLnHHcz:-KPp0iHPx0z0ImLnHHd-:b-uz0cbe) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L197)):*
> that needs to
that need to

---

*[GettingStartedWithGitHub.md, line 200 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoyVuvlyPo8JcCTZ8f:-KPoyVuvlyPo8JcCTZ8g:b-w0gk42) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L200)):*
> me
be

---

*[GettingStartedWithGitHub.md, line 201 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp1deWT_wUVtkIuK8B:-KPp1deWT_wUVtkIuK8C:bzb6m51) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L201)):*
> (origin)
Probably want to take out origin here since it could refer to either the upstream or fork's dev branches.

---

*[GettingStartedWithGitHub.md, line 233 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0xPCrL_7o9MbQJoD:-KPp0xPDyVPCI1fC-F6s:bvf2a52) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L233)):*
> In this example we can do it so the resulting `README.md` looks like this.
After resolving the conflict, `README.md` looks like this:

---

*[GettingStartedWithGitHub.md, line 250 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPowExtISTW7Oy3ANpj:-KPowExtISTW7Oy3ANpk:b-c8kd0z) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L250)):*
> you can 
can you

---

*[GettingStartedWithGitHub.md, line 256 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoyufNu0tvh8yVGF3y:-KPoyufORu6Wysadq7rg:bloj1bv) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L256)):*
> ```Markdown
> ```
> git status          # (optional) If you unsure of the name, you can use this to see the files that was in conflict.
> git add <file>      # Do this for each file that you fixed merged conflicts in. I.e 'git add README.md'.
> ```

May want to add a note here that you can also just do ```git *``` to add all un-added files at once.

---

*[GettingStartedWithGitHub.md, line 265 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoz9DqonjE5BIBvAmu:-KPoz9Dr3Lw4ikw0fT4_:b-xbiqyv) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L265)):*
> to your branch to your forked repository
to your branch in your forked repository

---

*[GettingStartedWithGitHub.md, line 281 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPozLYqtW6DTsNkxtvu:-KPozLYqtW6DTsNkxtvv:b-v005si) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L281)):*
> running tests, that folder
running tests. This folder

---

*[GettingStartedWithGitHub.md, line 294 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPozV1LSpiANe01I40S:-KPozV1LSpiANe01I40T:bc4570n) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L294)):*
> place
placed

---

*[GettingStartedWithGitHub.md, line 300 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPozjLOuXKUhyuGs9-R:-KPozjLOuXKUhyuGs9-S:b-w9zcac) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L300)):*
> that you point the the original pull
that you mention the original pull

---

*[GettingStartedWithGitHub.md, line 321 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp--Y6ERLyxaY6Kxa2:-KPp--Y6ERLyxaY6Kxa3:blpi2wh) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L321)):*
> ```Markdown
> ```
> git fetch <username>
> git branch -u <username>/<branch>
> ```

Again ```git branch -u ``` not necessary here

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177#-:-KPp1mjDkIwvdptVDuqA:b-8rhtcp)*
<!-- Sent from Reviewable.io -->

johlju commented:
Comments in code blocks seem to be to long for one page. Shall I fix those by adding the comment on two rows above the command instead? Or better idea, or just leave them as-is?
See here [GettingStartedWithGitHub](https://github.com/johlju/DscResources/blob/changes-to-gettingstartedwithgithub/GettingStartedWithGitHub.md)

---

Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 26 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 171 \[r3\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPKondi_aXhgJKf81J7:-KPs7ac72ZLLyVCUDe0D:b-6nu95j) ([raw file](https://github.com/powershell/dscresources/blob/bf3e9d89c11a159bdf34af753dc70577eca56ef7/GettingStartedWithGitHub.md#L171)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Wow. I just realized that origin in this documentation is actually the upstream branch...
> Normally git defaults to setting up origin as your fork, so I always think of 'origin' as your fork.
> Sorry if that caused any confusion.
> 
> 
> The ```git branch -u origin/dev``` step is not needed.
</details>
No worries, I figured that out pretty fast. I knew GitHub Desktop also did that, when I tried cloning from that, and it also named the upstream remote to 'PowerShell'. Very confusing for a beginner :) 
I learned git from this document, so for me the upstream will always and forever be 'origin' to me now :smile: 

Done.

---

*[GettingStartedWithGitHub.md, line 110 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp1QsPVQloefc5r3o5:-KPs6trYkjCXS9yUZ7bH:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L110)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > to your forks dev branch
> to your fork's dev branch
> 
</details>
Done

---

*[GettingStartedWithGitHub.md, line 111 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPowWduDW1V2n52oK3o:-KPs6lJjlZJJQO46ZTXL:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L111)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> This note is not needed. You don't need to set up a tracking branch to resolve merge conflicts.
</details>
Done

---

*[GettingStartedWithGitHub.md, line 158 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp-UtK64q7ufsfpeGT:-KPs71utMv_F33bX-UFR:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L158)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> >  merged while yours are in review,
> merged while yours is in review,
> > then you have to get those new changes into
> you will need to add those new changes into
> > before your pull request are allowed to merge
> before your pull request is allowed to merge
</details>
Done

---

*[GettingStartedWithGitHub.md, line 159 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp031auZMi9cHRe07D:-KPs73ERvf87K6sbvVWy:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L159)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > That
> This
</details>
Done

---

*[GettingStartedWithGitHub.md, line 162 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0ITcF9rlfCZ0Jrnb:-KPs7557ZnO2v-609b0_:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L162)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > It's a few steps to get this done.
> Here are the steps to rebasing your branch:
</details>
Done

---

*[GettingStartedWithGitHub.md, line 181 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0XUdxN-Kj5r18Qq5:-KPs9J_lM6gSifG1VrQc:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L181)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > needs
> need
</details>
Done

---

*[GettingStartedWithGitHub.md, line 183 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0bXFIRoDYoLcT6TO:-KPsB52fbZCxGH83V1Jf:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L183)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > your fork dev
> your fork's dev
</details>
Done

---

*[GettingStartedWithGitHub.md, line 194 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoyHb9AgcQX_9wKEaO:-KPs9iodIHBtyhZvizfB:bvt277z) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L194)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Again ```git branch -u origin/dev``` step is not needed.
</details>
Done. I removed `git branch -u my/dev'. 

---

*[GettingStartedWithGitHub.md, line 197 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0iHPx0z0ImLnHHcz:-KPs9umX30VwL4aMppSk:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L197)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > that needs to
> that need to
</details>
Done

---

*[GettingStartedWithGitHub.md, line 200 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoyVuvlyPo8JcCTZ8f:-KPs9vcJ_vZO0b4WBuF2:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L200)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > me
> be
</details>
Done

---

*[GettingStartedWithGitHub.md, line 201 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp1deWT_wUVtkIuK8B:-KPs9x9c4P73YpaucbjT:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L201)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > (origin)
> Probably want to take out origin here since it could refer to either the upstream or fork's dev branches.
</details>
Done

---

*[GettingStartedWithGitHub.md, line 233 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp0xPCrL_7o9MbQJoD:-KPsB3Buq1KuP4H2hfRo:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L233)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > In this example we can do it so the resulting `README.md` looks like this.
> After resolving the conflict, `README.md` looks like this:
</details>
Done

---

*[GettingStartedWithGitHub.md, line 250 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPowExtISTW7Oy3ANpj:-KPsA5uN8asMfE7fS0_P:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L250)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > you can 
> can you
> 
</details>
Done

---

*[GettingStartedWithGitHub.md, line 256 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoyufNu0tvh8yVGF3y:-KPsA8Z1HVniElPtX7Hz:bwdhhg3) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L256)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> May want to add a note here that you can also just do ```git *``` to add all un-added files at once.
</details>
Done. I choose 'stage' to give the beginner a new term to what 'add' also means. Can change it to add if that is better.

---

*[GettingStartedWithGitHub.md, line 265 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoz9DqonjE5BIBvAmu:-KPsAQNppz-LE7br_URd:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L265)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > to your branch to your forked repository
> to your branch in your forked repository
</details>
Done

---

*[GettingStartedWithGitHub.md, line 281 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPozLYqtW6DTsNkxtvu:-KPsARW-HW3ophPseZ6t:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L281)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > running tests, that folder
> running tests. This folder
</details>
Done

---

*[GettingStartedWithGitHub.md, line 294 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPozV1LSpiANe01I40S:-KPsATnPGaBBK4Cu0OqS:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L294)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > place
> placed
</details>
Done

---

*[GettingStartedWithGitHub.md, line 300 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPozjLOuXKUhyuGs9-R:-KPsAYInMKX49f1Qgcn9:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L300)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > that you point the the original pull
> that you mention the original pull
</details>
Done

---

*[GettingStartedWithGitHub.md, line 321 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPp--Y6ERLyxaY6Kxa2:-KPsAaEqgG8CpcK4b5bd:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L321)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Again ```git branch -u ``` not necessary here
</details>
Done

---

*[GettingStartedWithGitHub.md, line 109 \[r7\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPs6R7ZbfyFViHYvjHf:-KPs6R7ZbfyFViHYvjHg:bfi9zvz) ([raw file](https://github.com/powershell/dscresources/blob/140df0b792407b1ead0eadb6983ca7c7e98ef603/GettingStartedWithGitHub.md#L109)):*
> ```Markdown
> ## Making changes and pushing them to the fork
> 
> * To make changes, create a new local branch: `git checkout -b <branch> my/dev`, i.e. `git checkout -b awesome_feature my/dev`. 
> ```

I added `my/dev` after this since I thought tracking was necessary. Should I remove this? `git checkout -b <branch> my/dev`
Just want to make sure.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177#-:-KPs79w6NfhoABoqP8gw:b-twfx7a)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: all files reviewed at latest revision, 26 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 310 \[r8\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPsEOF-dm9LJYnYN-2V:-KPsEOF-dm9LJYnYN-2W:btnn5nt) ([raw file](https://github.com/powershell/dscresources/blob/fe50651d2aed790b710704626cbfd771d5635e5d/GettingStartedWithGitHub.md#L310)):*
> ```Markdown
> ```
> 
> 2. Add a remote to the original contributors fork. _In this example we use the users name as the remotes name._
> ```

This list, step 1-9 outputs as '1.'  on a all steps.  :/ Saw that when I looked at the .md file in my fork/branch.
Trying to solve this

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: 0 of 1 files reviewed at latest revision, 26 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 310 \[r8\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPsEOF-dm9LJYnYN-2V:-KPsHtYmUzV8FH0Av7li:bt2vr0y) ([raw file](https://github.com/powershell/dscresources/blob/fe50651d2aed790b710704626cbfd771d5635e5d/GettingStartedWithGitHub.md#L310)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> This list, step 1-9 outputs as '1.'  on a all steps.  :/ Saw that when I looked at the .md file in my fork/branch.
> Trying to solve this
</details>
Fixed!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: 0 of 1 files reviewed at latest revision, 27 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 340 \[r9\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPsJFLUvaKDi73--C5o:-KPsJFLUvaKDi73--C5p:btg76es) ([raw file](https://github.com/powershell/dscresources/blob/7c0c4f2a77a6315b8be28c684c8b555ef800bc9b/GettingStartedWithGitHub.md#L340)):*
> ```Markdown
> 
>  ```
>  git branch -u my/dev
> ```

Remove this line as well `git branch -u my/dev`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Almost there! Only two more tiny fixes :smile: 

---

Review status: 0 of 1 files reviewed at latest revision, 6 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 256 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoyufNu0tvh8yVGF3y:-KQMw7s4xWbculINt3r6:b-9zvpny) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L256)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Done. I choose 'stage' to give the beginner a new term to what 'add' also means. Can change it to add if that is better.
</details>
Oops I meant ```git add *``` instead of ```git *```. My bad. Sorry

---

*[GettingStartedWithGitHub.md, line 109 \[r7\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPs6R7ZbfyFViHYvjHf:-KQMvMPaybaZ0KG8nBKQ:bf06rnp) ([raw file](https://github.com/powershell/dscresources/blob/140df0b792407b1ead0eadb6983ca7c7e98ef603/GettingStartedWithGitHub.md#L109)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> I added `my/dev` after this since I thought tracking was necessary. Should I remove this? `git checkout -b <branch> my/dev`
> Just want to make sure.
</details>
You can leave this. It won't hurt and then we don't have to add a note about making sure you are in the correct branch before this command.

---

*[GettingStartedWithGitHub.md, line 340 \[r9\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPsJFLUvaKDi73--C5o:-KQMwWnv7rP6YP2C54Hl:b2nhtan) ([raw file](https://github.com/powershell/dscresources/blob/7c0c4f2a77a6315b8be28c684c8b555ef800bc9b/GettingStartedWithGitHub.md#L340)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary>
> Remove this line as well `git branch -u my/dev`?
</details>
Yup

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177#-:-KQMwZtisNhyrRoklVBQ:b-8woso)*
<!-- Sent from Reviewable.io -->

johlju commented:
I hope this is it then! :)

---

Reviewed 1 of 1 files at r10.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[GettingStartedWithGitHub.md, line 256 \[r6\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPoyufNu0tvh8yVGF3y:-KQR9TgB2ygW9c7dvUTT:b-731f9d) ([raw file](https://github.com/powershell/dscresources/blob/b29e14335eb8cd9ca2d03164fe027d633811545e/GettingStartedWithGitHub.md#L256)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Oops I meant ```git add *``` instead of ```git *```. My bad. Sorry
</details>
Ah, no worries, I wasn't even reading what I wrote either :) 

Done.

---

*[GettingStartedWithGitHub.md, line 109 \[r7\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPs6R7ZbfyFViHYvjHf:-KQR9H3KDT_sCpSJ1wlv:bghpthq) ([raw file](https://github.com/powershell/dscresources/blob/140df0b792407b1ead0eadb6983ca7c7e98ef603/GettingStartedWithGitHub.md#L109)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> You can leave this. It won't hurt and then we don't have to add a note about making sure you are in the correct branch before this command.
</details>
Done. Leaving as-is.

---

*[GettingStartedWithGitHub.md, line 340 \[r9\]](https://reviewable.io:443/reviews/powershell/dscresources/177#-KPsJFLUvaKDi73--C5o:-KQR9qaNM4iJvaYRCpwT:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/7c0c4f2a77a6315b8be28c684c8b555ef800bc9b/GettingStartedWithGitHub.md#L340)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Yup
</details>
Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177#-:-KQRAJ6qgtsfWaBHg_Qe:b9zdv6s)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r10.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/177#-:-KQRRsPxZT5KB2K3MzyN:bnfp4nl)*
<!-- Sent from Reviewable.io -->
